pypeit/reduce.py

```diff
@@ -528,35 +535,73 @@ def global_skysub(self, scaleImg=None, skymask=None, update_crmask=True, trim_ed
         # Mask objects using the skymask? If skymask has been set by objfinding, and masking is requested, then do so
         skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
 
-        # Select the edges to use: Selects the edges tweaked by the
-        # illumination profile if they're present; otherwise, it
-        # selects the original edges from EdgeTraceSet. To always
-        # select the latter, use the method with `original=True`.
-        # TODO -- JXP removed these from KCWI4..
-        #original = True if ((trim_edg[0] == 0) and (trim_edg[1] == 0)) else False
-        #left, right = self.slits.select_edges(original=original)
-        #left, right = self.slits.select_edges()
-
         if self.par['reduce']['skysub']['joint_fit']:
```

Comment:
As I think we are unlikely to use joint_fit in any other context except for IFUs, my suggestion is that you simply overload global skysub rather than adding this if then. Perhaps @profxj would like to comment. But we created this inheritance structure for exactly this reason.

Reply:
Some IFUs will not want to joint fit, so I would still need the if/then statement. There might be some cases (e.g. multislit which contains a narrow slit with very limited sky regions) that one might wish to perform a joint fit (at least for that slit). I agree otherwise, this is a pretty specific need... I've left it in for now unless there is resistance.

Reply:
I am not happy with it, because I don't agree that we would ever do this in multislit. We certainly have never had the need to. The whole point of local_skysub_extract is to deal with these narrow slits and model the sky and object simultaneously. So your usage is compeltely IFU specific and should be a child not an if/then.

Reply:
OK - I'll overload